src/devices/nm-acd-manager.c

```diff
@@ -450,18 +472,20 @@ nm_acd_manager_init (NMAcdManager *self)
 }
 
 NMAcdManager *
-nm_acd_manager_new (int ifindex, const guint8 *hwaddr, size_t hwaddr_len)
+nm_acd_manager_new (int ifindex, const guint8 *hwaddr, guint hwaddr_len)
 {
 	NMAcdManager *self;
 	NMAcdManagerPrivate *priv;
 
+	g_return_val_if_fail (ifindex > 0, NULL);
 	g_return_val_if_fail (hwaddr, NULL);
 	g_return_val_if_fail (hwaddr_len == ETH_ALEN, NULL);
 
 	self = g_object_new (NM_TYPE_ACD_MANAGER, NULL);
 	priv = NM_ACD_MANAGER_GET_PRIVATE (self);
 	priv->ifindex = ifindex;
-	memcpy (priv->hwaddr, hwaddr, ETH_ALEN);
+	priv->hwaddr = g_memdup (hwaddr, hwaddr_len);
```

Comment:
nm_memdup(). There is little difference in practice, but I think we should restrain ourself from API that is error prone.